plugin/Record/RecordManagerDuplicateListViewAddress.php

```diff
@@ -0,0 +1,136 @@
+{
+	/** @var wpdb */
+	private $_pWPDB;
+
+	/** @var Container */
+	private $_pContainer = null;
+
+	/**
+	 *
+	 * @param wpdb $pWPDB
+	 * @param Container $_pContainer
+	 * @throws \Exception
+	 */
+
+	public function __construct(wpdb $pWPDB, Container $_pContainer = null)
+	{
+		$this->_pWPDB = $pWPDB;
+		if ($_pContainer === null) {
+			$pDIContainerBuilder = new ContainerBuilder;
+			$pDIContainerBuilder->addDefinitions(ONOFFICE_DI_CONFIG_PATH);
+			$_pContainer = $pDIContainerBuilder->build();
+		}
+		$this->_pContainer = $_pContainer;
+	}
+
+	/**
+	 *
+	 * @param string $name
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 */
+
+	public function duplicateByName(string $name)
+	{
+		$prefix = $this->_pWPDB->prefix;
+
+		/* @var $pRecordManagerReadListViewAddress RecordManagerReadListViewAddress */
+		$pRecordManagerReadListViewAddress = $this->_pContainer->get(RecordManagerReadListViewAddress::class);
+		$listViewRoot = $pRecordManagerReadListViewAddress->getRowByName($name);
+
+		if (isset($listViewRoot) && $listViewRoot != null) {
+			$tableListViews = $prefix . self::TABLENAME_LIST_VIEW_ADDRESS;
+
+			$originalName = $listViewRoot['name'];
+			$newName = "{$originalName} - Copy";
+			$selectLikeOriginalName = "SELECT `name`, `listview_address_id` FROM {$this->_pWPDB->_escape($tableListViews)} WHERE name LIKE '{$this->_pWPDB->_escape($originalName)}%' ORDER BY listview_address_id DESC";
+			$listViewsRows = $this->_pWPDB->get_row($selectLikeOriginalName);
+			$id = $listViewsRows->listview_address_id;
+			if (!empty($listViewsRows->name) && $listViewsRows->name !== $originalName) {
+				preg_match('/\s[0-9]+$/', $listViewsRows->name, $matches);
+				$counter = 2;
+				if (!empty($matches)) {
+					$counter = (int)$matches[0] + 1;
+				}
+				$newName = "{$newName} $counter";
+			}
+
+			$newNameData = ['name' => $newName];
+
+			//duplicate root data to new list view
+			$newListView = [];
+			foreach ($listViewRoot as $key => $value) {
+				if (is_array($value) || $key === 'listview_address_id' || $key === 'name') {
+					continue;
+				}
+				$newListView[$key] = $value;
+			}
+			$newListView = array_merge($newNameData, $newListView);
+
+			$this->_pWPDB->insert(
+				$tableListViews,
+				$newListView
+			);
+			$duplicateListViewId = $this->_pWPDB->insert_id;
+
+			if ($duplicateListViewId !== 0) {
+				//duplicate data related oo_plugin_fieldconfig table
+				$tableFieldConfig = $prefix . self::TABLENAME_FIELDCONFIG_ADDRESS;
+				foreach ($listViewRoot['fields'] as $field) {
+					$selectFieldConfigByIdAndFieldName = "SELECT * FROM {$this->_pWPDB->_escape($tableFieldConfig)} WHERE listview_address_id='{$this->_pWPDB->_escape($id)}' AND fieldname ='{$this->_pWPDB->_escape($field)}'";
+					$fieldConfigRows = $this->_pWPDB->get_results($selectFieldConfigByIdAndFieldName);
+
+
+					if (!empty($fieldConfigRows) && (count($fieldConfigRows) !== 0)) {
+						foreach ($fieldConfigRows as $fieldConfigRow) {
```

Comment:
Wouldn't it be easier to copy the entire `$fieldConfigRow` and strip the fields one doesn't want (such as ID). Basically, to have a disallow-list of fields, instead of an allow-list.

Reply:
You're right, I'll do it